src/Lexer.php

```diff
@@ -42,11 +53,17 @@ class Lexer
     private int $lineNumber;
     private bool $eos;
     /**
-     * @var array<string, string>
+     * @phpstan-var array<TKeywordsType, TKeywordsString>
      */
     private array $keywordsCache = [];
     /**
-     * @var array<string, list<string>>
+     * @var array{}|array{
```

Comment:
This property is refactored entirely in #350

Reply:
Are you suggesting that I undo this change?

Reply:
I would say yes. It will avoid conflicts between the 2 PRs.